libethcore/SealEngine.h

```diff
@@ -99,8 +99,11 @@ class SealEngineBase: public SealEngineFace
 public:
 	void generateSeal(BlockHeader const& _bi) override
 	{
+        BlockHeader header(_bi);
```

Comment:
I think this should happen in `NoProof`, not in `SealEngineBase`
Also I don't see any reason for `SealEngineBase` to have `generateSeal` implementation at all, so try to just move it to `NoProof`